.circleci/config.yml

```diff
@@ -9,89 +12,87 @@ executors:
       - add_ssh_keys:
           fingerprints:
-            - '80:24:c0:40:28:36:f1:80:7f:fd:15:c3:09:cf:86:fc'
+            - "80:24:c0:40:28:36:f1:80:7f:fd:15:c3:09:cf:86:fc"
       - attach_workspace:
           at: /home/circleci
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
-          name: 'Build and deploy docs'
+          name: Build and deploy docs
           command: |
-            chmod +x .circleci/add-known-hosts.sh
             poetry run ./.circleci/add-known-hosts.sh
-            chmod +x .circleci/github-pages.sh
             poetry run ./.circleci/github-pages.sh
+          environment:
+            REPOSITORY_PATH: /home/circleci/docs
+            DEPLOYMENT_GIT_SSH: git@github.com:Sceptre/sceptre.github.io.git
 
 jobs:
   build:
-    docker:
-      - image: sceptreorg/sceptre-circleci:2.1.0
+    executor: python/default
     steps:
       - checkout
       - restore_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
-      - run:
-          name: 'Installing Dependencies'
-          command: poetry install --all-extras -v
+          key: sceptre-v1-{{ checksum ".pre-commit-config.yaml" }}
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
           name: Run pre-commit
           command: poetry run pre-commit run --all-files --show-diff-on-failure
       - run:
           name: Build package
           command: poetry build
       - save_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
+          key: sceptre-v1-{{ checksum ".pre-commit-config.yaml" }}
           paths:
-            - '~/.cache/pypoetry'
-            - '~/.cache/pre-commit'
-            - '.tox'
+            - ~/.cache/pre-commit
 
   unit-tests:
-    docker:
-      - image: sceptreorg/sceptre-circleci:2.1.0
+    parameters:
+      python:
+        type: string
+    executor:
+      name: python/default
+      tag: <<parameters.python>>
     steps:
       - checkout
       - restore_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
+          key: sceptre-v1-{{ checksum "poetry.lock" }}-{{ checksum "tox.ini" }}
+      # Update poetry for https://github.com/python-poetry/poetry/issues/7184
+      - run: poetry self update --no-ansi
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
           name: Unit Tests
           command: |
-            poetry run tox -e clean
-            poetry run tox
+            poetry run tox -e py$(echo <<parameters.python>> | tr -d .)
```

Comment:
would be great if we can add some comments here describing why this is needed. it would help give context to others debugging or updating the CI workflow.

Reply:
[See my comment suggestion on the line just above](https://github.com/Sceptre/sceptre/pull/1395/files/b6fa015e4c430eda7252e1f7839bf55db5662fab#r1433045343).